massa-client/src/repl.rs

```diff
@@ -135,20 +136,48 @@ impl Completion for CommandCompletion {
     /// Simple completion implementation based on substring
     fn get(&self, input: &str) -> Option<String> {
         let input = input.to_string();
-        let suggestions: Vec<&str> = self
-            .options
-            .iter()
-            .filter(|s| s.len() >= input.len() && input == s[..input.len()])
-            .map(|s| &s[..])
-            .collect();
-        if !suggestions.is_empty() {
-            println!();
-            for suggestion in &suggestions {
-                println!("{}", style(suggestion).dim());
+        if input.contains(' ') {
+            let mut args: Vec<&str> = input.split(' ').collect();
+            let mut default_path = "./";
+            let path_to_complete = args.last_mut().unwrap_or(&mut default_path);
+            if let Ok(paths) = glob(&(path_to_complete.to_owned() + "*")) {
+                let suggestions: Vec<String> = paths
+                    .filter_map(|x| {
+                        if let Ok(path) = x {
+                            Some(path.display().to_string())
+                        } else {
+                            None
+                        }
```

Comment:
Here you maybe could use `x.map(|path| path.display().to_string())` for less verbosity

Reply:
Agree !